### PR TITLE
[Moey Sean Jean] iP

### PR DESCRIPTION
**Pull request by MoeySeanJean**
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2526s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

### CODE REVIEW

File: src/test/java/jaiden/task/TaskListTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    @Test
+    public void showTest() {
+        assertEquals("     Here are the tasks on Aug 22 2025 in your list:\n", new TaskList().show(LocalDate.parse("2025-08-22")));
+        ArrayList<Task> tasks = new ArrayList<>();
+        tasks.add(new Deadline("test", LocalDate.parse("2025-08-22")));
+        tasks.add(new Event("test", LocalDate.parse("2025-08-22"), LocalDate.parse("2025-08-22")));
+        tasks.add(new Deadline("test", LocalDate.parse("2024-08-22")));
+        tasks.add(new Event("test", LocalDate.parse("2024-08-22"), LocalDate.parse("2024-08-22")));
+        assertEquals("     Here are the tasks on Aug 22 2025 in your list:\n     1.[D][ ] test (by: Aug 22 2025)\n     2.[E][ ] test (from: Aug 22 2025 to: Aug 22 2025)\n", new TaskList(tasks).show(LocalDate.parse("2025-08-22")));
```
Comment (ChekaLowQiJun) [inline src/test/java/jaiden/task/TaskListTest.java]: Maybe you could consider splitting this line into two separate lines with the appropriate indentation level? 
```suggestion
        assertEquals("     Here are the tasks on Aug 22 2025 in your list:\n     1.[D][ ] test (by: Aug 22 2025)\n     2.[E][ ] test (from: Aug 22 2025 to: Aug 22 2025)\n", new TaskList(tasks).show(LocalDate.parse("2025-08-22")));
```

File: src/test/java/jaiden/ui/ParserTest.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package jaiden.ui;
+
+import jaiden.command.*;
```
Comment (ChekaLowQiJun) [inline src/test/java/jaiden/ui/ParserTest.java]: Perhaps a wildcard should not be used in the import statement here?
```suggestion
import jaiden.command.*;
```

File: src/test/java/jaiden/task/EventTest.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package jaiden.task;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (ChekaLowQiJun) [inline src/test/java/jaiden/task/EventTest.java]: Perhaps reconsider the ordering of your import statements to be consistent and ensure they comply with the Java coding standards?
```suggestion
import static org.junit.jupiter.api.Assertions.assertEquals;
```

File: src/test/java/jaiden/storage/StorageTest.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        Storage storage = new Storage("data/test.txt");
+        ArrayList<Task> test = new ArrayList<>();
+        test.add(new Todo("test"));
+        test.add(new Deadline("test", LocalDate.parse("2025-08-22")));
+        test.add(new Event("test", LocalDate.parse("2025-08-22"), LocalDate.parse("2025-08-22")));
+        test.add(new Todo("test", true));
+        test.add(new Deadline("test", true, LocalDate.parse("2025-08-22")));
+        test.add(new Event("test", true, LocalDate.parse("2025-08-22"), LocalDate.parse("2025-08-22")));
+        storage.save(new TaskList(test));
+        assertEquals("T | 0 | test\nD | 0 | test | 2025-08-22\nE | 0 | test | 2025-08-22 | 2025-08-22\nT | 1 | test\nD | 1 | test | 2025-08-22\nE | 1 | test | 2025-08-22 | 2025-08-22", new Scanner(new File("data/test.txt")).useDelimiter("\\Z").next());
```
Comment (ChekaLowQiJun) [inline src/test/java/jaiden/storage/StorageTest.java]: Should this line be this long? 
```suggestion
        assertEquals("T | 0 | test\nD | 0 | test | 2025-08-22\nE | 0 | test | 2025-08-22 | 2025-08-22\nT | 1 | test\nD | 1 | test | 2025-08-22\nE | 1 | test | 2025-08-22 | 2025-08-22", new Scanner(new File("data/test.txt")).useDelimiter("\\Z").next());
```

File: src/test/java/jaiden/command/DeleteCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package jaiden.command;
+
+import jaiden.storage.Storage;
+import jaiden.task.Task;
+import jaiden.task.TaskList;
+import jaiden.task.Todo;
+import jaiden.ui.Ui;
+import org.junit.jupiter.api.Test;
```
Comment (ChekaLowQiJun) [inline src/test/java/jaiden/command/DeleteCommandTest.java]: Can the import statements be categorised into better logical blocks?
```suggestion
import org.junit.jupiter.api.Test;
```

File: src/main/java/jaiden/ui/Parser.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        case "event":
+            return new AddCommand(commands);
+        case "delete":
+            return new DeleteCommand(commands);
+        case "bye":
+            return new ExitCommand(commands);
+        case "find":
+            return new FindCommand(commands);
+        default:
+            return new UnknownCommand(commands);
```
Comment (ChekaLowQiJun) [inline src/main/java/jaiden/ui/Parser.java]: The way you handle the cases is very clean!
```suggestion
        case "list":
        case "show":
            return new ListCommand(commands);
        case "mark":
        case "unmark":
            return new MarkCommand(commands);
        case "todo":
        case "deadline":
        case "event":
            return new AddCommand(commands);
        case "delete":
            return new DeleteCommand(commands);
        case "bye":
            return new ExitCommand(commands);
        case "find":
            return new FindCommand(commands);
        default:
            return new UnknownCommand(commands);
```

File: src/main/java/jaiden/command/AddCommand.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+            task = new Deadline(description, by);
+        } else {
+            LocalDate from = LocalDate.parse(commands[3]);
+            LocalDate to = LocalDate.parse(commands[5]);
+            task = new Event(description, from, to);
+        }
+        String msg = taskList.add(task);
+        ui.show(msg);
+        storage.save(taskList);
+    }
```
Comment (ChekaLowQiJun) [inline src/main/java/jaiden/command/AddCommand.java]: Is there a way to add line breaks and categorise the code into logical blocks to increase readability?

```suggestion
    public void execute(TaskList taskList, Ui ui, Storage storage) {
        String description = commands[1];
        Task task;
        if (commands[0].equals("todo")) {
            task = new Todo(description);
        } else if (commands[0].equals("deadline")) {
            LocalDate by = LocalDate.parse(commands[3]);
            task = new Deadline(description, by);
        } else {
            LocalDate from = LocalDate.parse(commands[3]);
            LocalDate to = LocalDate.parse(commands[5]);
            task = new Event(description, from, to);
        }
        String msg = taskList.add(task);
        ui.show(msg);
        storage.save(taskList);
    }
```

**Review by ChekaLowQiJun — COMMENTED** [review]
Overall your codebase is really neat and readable, albeit certain run-on lines of code and inappropriate ordering of statements. Good job thus far for you project and keep up the good work!

File: src/test/java/jaiden/command/AddCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,75 @@
+package jaiden.command;
```
Comment (dlywlotus) [inline src/test/java/jaiden/command/AddCommandTest.java:1]: Good comprehensive testing!

File: src/test/java/jaiden/task/DeadlineTest.java [hunk 9]
```diff
@@ -0,0 +1,25 @@
+package jaiden.task;
```
Comment (dlywlotus) [inline src/test/java/jaiden/task/DeadlineTest.java:1]: Once again, the good tests that take into account possible edge cases.

File: src/test/java/jaiden/ui/JaidenTest.java [hunk 10]
```diff
@@ -0,0 +1,90 @@
+package jaiden.ui;
```
Comment (dlywlotus) [inline src/test/java/jaiden/ui/JaidenTest.java:1]: Same here regarding the testing practices! :))))

**Review by dlywlotus — COMMENTED** [review]
Overall, the code is well structured and well tested. LGTM!
